projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -0,0 +1,875 @@
+    public registerOnValidatorChange?(fn: any): void {
+        this.onValidatorChange = fn;
+    }
+
+    /** @hidden @internal */
+    public setDisabledState?(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+    }
+
+    /** @hidden @internal */
+    get separatorClass(): string {
+        return this.getComponentDensityClass('igx-date-range-picker__label');
+    }
+
+    /** @hidden */
+    public ngOnInit(): void {
+        this._ngControl = this._injector.get<NgControl>(NgControl, null);
+    }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        switch (this.mode) {
+            case InteractionMode.DropDown:
+                this.attachOnKeydown();
+                this.subscribeToDateEditorEvents();
+                break;
+        }
+        this.configPositionStrategy();
+        this.configOverlaySettings();
+        this.attachOnTouched();
+
+        const subsToClicked = () => {
+            this.$toggleClickNotifier.next();
+            this.toggleComponents.forEach(toggle => {
+                toggle.clicked.pipe(takeUntil(this.$toggleClickNotifier)).subscribe(() => this.open());
+            });
+        };
+        this.toggleComponents.changes.pipe(takeUntil(this.$destroy)).subscribe(() => subsToClicked());
+        subsToClicked();
+
+        this.setRequiredToInputs();
+        // detect changes?
+
+        if (this._ngControl) {
+            this._statusChanges$ = this._ngControl.statusChanges.subscribe(this.onStatusChanged.bind(this));
+        }
+        this.updateInputs();
+    }
+
+    /** @hidden @internal */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes['locale']) {
+            this.inputFormat = DatePickerUtil.getDefaultInputFormat(this.locale || 'en') || DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public ngOnDestroy(): void {
+        this.$destroy.next();
+        this.$destroy.complete();
+        this.$toggleClickNotifier.next();
+        this.$toggleClickNotifier.complete();
+    }
+
+    /** @hidden @internal */
+    public handleOpening(event: CancelableEventArgs): void {
+        this.onOpening.emit(event);
+    }
+
+    /** @hidden @internal */
+    public handleOpened(): void {
+        requestAnimationFrame(() => {
+            this.calendar.daysView.focusActiveDate();
+            this.onOpened.emit({ owner: this });
+        });
+    }
+
+    /** @hidden @internal */
+    public handleClosing(event: CancelableEventArgs): void {
+        this.onClosing.emit(event);
+    }
+
+    /** @hidden @internal */
+    public handleClosed(): void {
+        // TODO: move to func
+        if (this.value && this.value.start && !this.value.end) {
+            this.value = { start: this.value.start, end: this.value.start };
+        }
+        if (this.value && !this.value.start && !this.value.end) {
+            this.value = null;
+        }
+        if (this.value && !this.valueInRange(this.value)) {
```

Comment:
there is no way, except if `IgxCalendar` is broken, to select dates out of range. I would remove this check.